### DISCUSSION TITLE
Cache builds up 2-3 times every time, how to disable? Favicons folder too...

### DISCUSSION THREAD

**Discussion by vdbhb59** [FreshRSS/FreshRSS#6063] [Q&A] [answered]
Just need one help please. The /data/cache folder builds up a lot and takes up almost 3 times the space of total itself. How do I disable the cache? If possible, should I disable the cache, or would it increase the load on my resources?
Also, the favicons folder also builds up. Is there a way to stop/minimize that too please?
I have tried searching, but unable to find a solution.
Also, I have shared hosting, with jail access and not root access.

**Comment by Alkarex** [comment]
What sizes are you observing, and for how many users and feeds?
Disabling the cache would considerably increase the resource consumption on upstream servers.
What do you mean by *3 times the space of total itself*? Which total?

**Comment by vdbhb59** [comment]
> What sizes are you observing, and for how many users and feeds? Disabling the cache would considerably increase the resource consumption on upstream servers. What do you mean by _3 times the space of total itself_? Which total?

So the total size of the feeds is example 4-5GB with 900+ users, but the cache is anywhere between 13-15 GB over a period of 3-5 days. Number of feeds is difficult to observe, but give or take 5-10 subscriptions per user, at a minimum level.
Yeah, I was aware that the resources would be very high, hence am not willing to go there at all.
Any idea how this can be minimized? Any settings/configuration option to minimize the cache held over can help. My hosting provider is going to change the total size available for my plan and minimize it which means I need to take care of my space, and hence cache comes to play a big role here.
Currently I am deleting cache every 7 days, as I do not get time every day. Cron job may help, but again deleting daily also will push the resource limit at least daily, which is another thing I want to avoid being shared hosting.

**Comment by Frenzie** [comment]
It's likely true that for our purposes we don't need a "cache" at all, just the last-modified date. It's been a while since I dove into SimplePie but I don't think that's easily doable though.

**Comment by vdbhb59** [comment]
> It's likely true that for our purposes we don't need a "cache" at all, just the last-modified date. It's been a while since I dove into SimplePie but I don't think that's easily doable though.

Yeah, I remember in one of my other tickets, Alkarex did [mention](https://github.com/FreshRSS/FreshRSS/issues/5771#issuecomment-1784052559) about some changes in `DATA_FREE` which lead to cache building up, but that was mostly with database. Still this cache is eating up a lot of space. I had to limit the number of registrations for the time being, as I am having 15+ new registrations daily.

Any solution would be great, a permanent one hopefully. :)

**Comment by Alkarex** [comment, marked as answer]
You can already tweak `CLEANCACHE_HOURS` in `./constants.local.php` as explained in:

https://github.com/FreshRSS/FreshRSS/blob/969758a73e37dd8540a2ffb41776b39e20ac44be/constants.php#L69-L70

You could consider going down to ~48 hours.

The local cache is not only used to avoid re-downloading unchanged feeds, but also to share feeds between users, when more than one user is following a given feed.

You can also count how much space is consumed by `./data/cache/*.html` (Web scraping or full content retrieval) vs. `./data/cache/*.spc` (normal RSS / ATOM feeds).

Favicons can be disabled at user level (but this decreases the quality of the FreshRSS user interface), with the `show_favicons` setting (which is mostly meant for users with 1000+ feeds or with very low bandwidth). Pruning the favicons cache means that the favicons will have to be re-downloaded again (which requires more than one network request, because it first needs to download the HTML homepage to discover the actual favicon).

The total size looks reasonable to me, given the large number of users. Reducing cache means an increase of network traffic, including an increased risk of getting banned from various Web sites (by IP, or worse for the community, by user-agent ). You consider calculating the maximum number of users you will be able to handle with the hardware you can afford. 1000+ users is optimistic for a shared hosting.

See our CLI for statistics and commands, especially useful when there are many users:
https://github.com/FreshRSS/FreshRSS/blob/edge/cli/README.md

**Reply by Alkarex** [reply]
Related discussion, about the need for a cache with a long enough duration
https://github.com/FreshRSS/FreshRSS/issues/6056

**Comment by Alkarex** [comment]
P.S. You can also tweak `max_inactivity` in your `./data/config.php` to avoid refreshing feeds of inactive users

https://github.com/FreshRSS/FreshRSS/blob/969758a73e37dd8540a2ffb41776b39e20ac44be/config.default.php#L108-L110

**Comment by Frenzie** [comment]
> The local cache is not only used to avoid re-downloading unchanged feeds, but also to share feeds between users, when more than one user is following a given feed.

Ah right, I forgot about that. My bad.

**Comment by vdbhb59** [comment]
> P.S. You can also tweak `max_inactivity` in your `./data/config.php` to avoid refreshing feeds of inactive users
> 
> https://github.com/FreshRSS/FreshRSS/blob/969758a73e37dd8540a2ffb41776b39e20ac44be/config.default.php#L108-L110

This depends on the php.ini value under PHP_INT_MAX correct?
Edit: The value was set to 9223372036854775807, so I changed it to 40 days = 3456000 in seconds. Hopefully that will help as well.

**Comment by vdbhb59** [comment]
> You could consider going down to ~48 hours.

I went with 72 hours for the time being.
